lib/compat/wordpress-6.0/block-template-utils.php

```diff
@@ -99,10 +99,22 @@ function gutenberg_generate_block_templates_export_file() {
 	// Load theme.json into the zip file.
 	$tree = WP_Theme_JSON_Resolver_Gutenberg::get_theme_data();
 	$tree->merge( WP_Theme_JSON_Resolver_Gutenberg::get_user_data() );
+	$theme_json_raw = $tree->get_data();
+	// If a version is defined, add a schema.
+	if ( $theme_json_raw['version'] ) {
+		global $wp_version;
+		$theme_json_version = substr( $wp_version, 0, strpos( $wp_version, '-' ) );
+		if ( defined( 'IS_GUTENBERG_PLUGIN' ) ) {
+			$theme_json_version = 'trunk';
+		}
+		$theme_json_raw = array_reverse( $theme_json_raw );
+		$theme_json_raw['$schema'] = 'https://schemas.wp.org/wp/' . $theme_json_version . '/theme.json';
```

Comment:
Can't `$schema` already exist? if it's added manually we should respect that.

Reply:
It's not stored in the tree - we could get it from the file directly, which is the approach I took at first (https://github.com/WordPress/gutenberg/pull/39590) but it seems like the recommended approach is to use the data from the class instance rather than getting it from the file, so it's properly sanitised etc.